aicsimageio/readers/nd2_reader.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Tuple
+
+from fsspec.implementations.local import LocalFileSystem
+
+from .. import constants, exceptions, types
+from ..utils import io_utils
+from ..utils.cached_property import cached_property
+from ..utils.dask_proxy import DaskArrayProxy
+from .reader import Reader
+
+if TYPE_CHECKING:
+    import xarray as xr
+    from fsspec.spec import AbstractFileSystem
+    from ome_types import OME
+
+
+try:
+    import nd2
+except ImportError:
+    raise ImportError(
+        "The nd2 package is required for this reader. "
+        "Install with `pip install aicsimageio[nd2]`"
+    )
+
+
+class ND2Reader(Reader):
+    """Read NIS-Elements files using the Nikon nd2 SDK.
+
+    This reader requires `nd2` to be installed in the environment.
+
+    Parameters
+    ----------
+    image : Path or str
+        path to file
+
+    Raises
+    ------
+    exceptions.UnsupportedFileFormatError
+        If the file is not supported by bioformats.
+    """
+
+    @staticmethod
+    def _is_supported_image(fs: AbstractFileSystem, path: str, **kwargs: Any) -> bool:
+        from nd2._util import NEW_HEADER_MAGIC_NUM
+
+        with fs.open(path, "rb") as fh:
+            return int.from_bytes(fh.read(4), "little") == NEW_HEADER_MAGIC_NUM
+
+    def __init__(self, image: types.PathLike):
+        self._fs, self._path = io_utils.pathlike_to_fs(image, enforce_exists=True)
+        # Catch non-local file system
+        if not isinstance(self._fs, LocalFileSystem):
+            raise ValueError(
+                f"Cannot read Bioformats from non-local file system. "
+                f"Received URI: {self._path}, which points to {type(self._fs)}."
+            )
+
+        if not self._is_supported_image(self._fs, self._path):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self._path
+            )
+
+    @property
+    def scenes(self) -> Tuple[str, ...]:
+        return ("Image:0",)  # TODO
```

Comment:
So bioformats reader supports multi-scene but this reader doesn't? Technically this is a "regression" because of the reader order in the formats dict. Although neither this or bioformats is released yet. So.... :shrug: